src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -680,7 +680,9 @@ private enum ActivationInitializationStage
                                 SchedulingContext).Ignore();
                         }
 
-                        RerouteAllQueuedMessages(activation, null, "Failed InvokeActivate", exception);
+                        // Reject all of the messages queued for this activation.
+                        var activationFailedMsg = nameof(Grain.OnActivateAsync) + " failed";
+                        RejectAllQueuedMessages(activation, activationFailedMsg, exception);
```

Comment:
Isn't there a race here that will allow new requests to get enqueue right after we reject all messages that are already in the queue? Are we preventing enqueuing of new requests before we start the rejection loop?